lib/chef/resource/powershell_package_source.rb

```diff
@@ -110,16 +109,16 @@ class PowershellPackageSource < Chef::Resource
         if package_source_exists?
           unregister_cmd = "Get-PackageSource -Name '#{new_resource.source_name}' | Unregister-PackageSource"
           converge_by("unregister source: #{new_resource.source_name}") do
-            res = powershell_out(unregister_cmd)
-            raise "Failed to unregister #{new_resource.source_name}: #{res.stderr}" unless res.stderr.empty?
+            res = powershell_exec(unregister_cmd)
+            raise "Failed to unregister #{new_resource.source_name}: #{res.errors}" if res.error?
           end
         end
       end
 
       action_class do
         def package_source_exists?
-          cmd = powershell_out!("(Get-PackageSource -Name '#{new_resource.source_name}' -WarningAction SilentlyContinue).Name")
-          cmd.stdout.downcase.strip == new_resource.source_name.downcase
+          cmd = powershell_exec!("(Get-PackageSource -Name '#{new_resource.source_name}' -ErrorAction SilentlyContinue).Name")
+          !cmd.result.empty? && cmd.result.to_s.downcase.strip == new_resource.source_name.downcase
```

Comment:
```suggestion
          String(cmd.result).downcase.strip == new_resource.source_name.downcase
```

Reply:
the problem here is that `String(cmd.result)` will return `"{}"` if it is empty.